src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -101,6 +101,32 @@ protected void AddEndpointsToEnvironment()
             }
         }
 
+        protected void AddSecureFilesToEnvironment() 
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(ExecutionContext, nameof(ExecutionContext));
+            ArgUtil.NotNull(ExecutionContext.SecureFiles, nameof(ExecutionContext.SecureFiles));
+
+            // Add the secure files to the environment variable dictionary.
+            foreach (SecureFile secureFile in ExecutionContext.SecureFiles)
+            {
+                ArgUtil.NotNull(secureFile, nameof(secureFile));
+                
+                if (secureFile.Id > 0)
```

Comment:
Even ID check is not required but since we are using it as the key, I wanted to make sure. The rest of the data is verified on the server before passing down to the agent.